pkg/utils/handler.go

```diff
@@ -79,36 +103,78 @@ func (h *Handler) AddMiddleware() {
 }
 
 func (h *Handler) UpdateAgents(rw http.ResponseWriter, r *http.Request, rp httprouter.Params) {
-	agentData := AgentInfo{}
+	agentData := ext_v1.AgentSpec{}
+
 	if err := ProcessRequest(r, &agentData, rw); err != nil {
 		return
 	}
 
 	agentData.LastUpdated = time.Now()
-	glog.V(10).Infof("Updating the agents cache with value: %v", agentData)
+	glog.V(10).Infof("Updating the agents resource with value: %v", agentData)
+
 	agentName := rp.ByName("name")
-	if _, exists := h.AgentCache[agentName]; !exists {
+	_, err := h.ExtensionsClientset.Agents().Get(agentName)
+
+	if err != nil {
+		glog.Error(err)
+	}
+
+	agent := &ext_v1.Agent{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name: agentName,
+		},
+		Spec: agentData,
+	}
+
+	if api_errors.IsNotFound(err) {
 		h.Metrics[agentName] = NewAgentMetrics(&agentData)
```

Comment:
Need to check, whether we have h.Metrics[agentName] already (to not loose metrics after restart of a server).